tests/webproxy.nix

```diff
@@ -25,18 +31,21 @@ import ./make-test-python.nix ({ pkgs, ... }:
           gateways = {};
         };
 
+        systemd.services.helloserver = {
+          wantedBy = [ "multi-user.target" ];
+          script = ''
+            echo 'Hello World!' > hello.txt
+            ${pkgs.python3.interpreter} -m http.server ${builtins.toString serverport} >&2
+          '';
+        };
       };
   };
   testScript = ''
     webproxy.wait_for_unit("varnish.service")
     webproxy.wait_for_unit("varnishncsa.service")
+    webproxy.wait_for_unit("helloserver.service")
 
-    webproxy.execute("""
-        echo 'Hello World!' > hello.txt
-        ${pkgs.python3.interpreter} -m http.server 8080 >&2 &
-        """)
-
-    url = 'http://localhost:8008/hello.txt'
+    url = 'http://localhost:${builtins.toString varnishport}/hello.txt'
```

Comment:
Ok, I think we should cycle through a few specialisations to show that adding/changing/removing vhosts works.

Reply:
definitely, will look into it